extensions/arc/src/common/utils.ts

```diff
@@ -66,3 +66,52 @@ export function getConnectionModeDisplayText(connectionMode: string | undefined)
 	}
 	return connectionMode;
 }
+
+/**
+ * Opens an input box prompting the user to enter in the name of a resource to delete
+ * @param namespace The namespace of the resource to delete
+ * @param name The name of the resource to delete
+ * @returns Promise resolving to true if the user confirmed the name, false if the input box was closed for any other reason
+ */
+export async function promptForResourceDeletion(namespace: string, name: string): Promise<boolean> {
+	const inputBox = vscode.window.createInputBox();
+	inputBox.title = loc.resourceDeletionWarning(namespace, name);
+	inputBox.placeholder = name;
+	return new Promise(resolve => {
+		let valueAccepted = false;
+		inputBox.show();
```

Comment:
Should it setup the onDid callbacks before showing the input box?  They could type really fast and submit before the callbacks are setup.

Reply:
D'oh, missed seeing these before hitting submit. I'll get them in another PR.